urlchecker/core/worker.py

```diff
@@ -88,8 +88,7 @@ def run(self, funcs: dict, tasks: dict) -> Optional[dict]:
 
         except:
             logger.error("Error running task")
-            pool.terminate()
-            sys.exit(1)
+            return None
```

Comment:
Sorry I wasn't clear. When we return at the bottom this already happens. Here you don't need to do anything except for logger.error.